tests/multiprocessing_smoke_tests.py

```diff
@@ -12,62 +13,65 @@
                                 AnnealingBoxCurriculum)
 from syllabus.tests import (create_cartpole_env,
                             create_nethack_env,
-                            get_test_values,
-                            test_native_multiprocess,
-                            test_ray_multiprocess,
-                            test_single_process)
+                            get_test_values)
+
+from syllabus.tests import test_single_process as run_single_process
+from syllabus.tests import test_native_multiprocess as run_native_multiprocess
+from syllabus.tests import test_ray_multiprocess as run_ray_multiprocess
 
 N_ENVS = 2
 N_EPISODES = 2
 
-if __name__ == "__main__":
-    ray.init()
-    nethack_env = create_nethack_env()
-    cartpole_env = create_cartpole_env()
-    curricula = [
+
+ray.init()
+nethack_env = create_nethack_env()
+cartpole_env = create_cartpole_env()
+
+curricula = [
         (NoopCurriculum, create_nethack_env, (NetHackScore, nethack_env.task_space), {}),
         (DomainRandomization, create_nethack_env, (nethack_env.task_space,), {}),
         # (LearningProgressCurriculum, create_nethack_env, (nethack_env.task_space,), {}),
         (CentralizedPrioritizedLevelReplay, create_nethack_env, (nethack_env.task_space,), {"device": "cpu", "suppress_usage_warnings": True, "num_processes": N_ENVS}),
-        (PrioritizedLevelReplay, create_nethack_env, (nethack_env.task_space, nethack_env.observation_space), {
-            "get_value": get_test_values,
-            "device": "cpu",
-            "num_processes": N_ENVS,
-            "num_steps": 2048
-        }),
+        #(PrioritizedLevelReplay, create_nethack_env, (nethack_env.task_space, nethack_env.observation_space), {
+        #    "get_value": get_test_values,
+        #    "device": "cpu",
+        #    "num_processes": N_ENVS,
+        #    "num_steps": 2048
+        #}),
         (SimpleBoxCurriculum, create_cartpole_env, (cartpole_env.task_space,), {}),
         (AnnealingBoxCurriculum, create_cartpole_env, (cartpole_env.task_space,), {
             'start_values': [-0.02, 0.02],
             'end_values': [-0.3, 0.3],
             'total_steps': [10]
         }),
     ]
-    for curriculum, env_fn, args, kwargs in curricula:
-        print("")
-        print("*" * 80)
-        print("Testing curriculum:", curriculum.__name__)
-        print("*" * 80)
-        print("")
 
-        # Test single process speed
-        print("RUNNING: Python single process test (1 env)...")
-        single_kwargs = kwargs.copy()
-        if "num_processes" in single_kwargs:
-            single_kwargs["num_processes"] = 1
-        test_curriculum = curriculum(*args, **single_kwargs)
-        native_speed = test_single_process(env_fn, curriculum=test_curriculum, num_envs=1, num_episodes=N_EPISODES)
-        print(f"PASSED: single process test (1 env) passed: {native_speed:.2f}s")
+@pytest.mark.parametrize("curriculum, env_fn, args, kwargs", curricula)
```

Comment:
This is really nice! I looked into this annotation a bit more and it looks like you can pass id names. So if you change this to:
```suggestion
@pytest.mark.parametrize("curriculum, env_fn, args, kwargs", curricula, ids=[test_names])
```

And somewhere above you define test_names like this:
```
test_names = [curriculum_args[0].__name__ for curriculum_args in curricula] 
```

It should be a bit easier to read. You should also change this for the other test functions.

Reply:
Point noted. Will be updated in next request along with `multiprocessing_speed_tests` by this weekend.